scopesim/effects/metis_wcu/fpmask.py

```diff
@@ -0,0 +1,125 @@
+"""A class for the METIS WCU focal-plane mask"""
+
+from pathlib import Path
+import numpy as np
+from matplotlib import pyplot as plt
+from astropy.io import fits
+from astropy import units as u
+from ..data_container import DataContainer
+from ...utils import find_file, from_currsys, get_logger
+from ...optics.image_plane_utils import sub_pixel_fractions
+
+logger = get_logger(__name__)
+
+class FPMask:
+    """Focal-plane mask for the METIS WCU
+
+    Parameters
+    ----------
+    See :class:`DataContainer` for input parameters
+
+    """
+    hdr = {"BG_SRC": True,
+           "BG_SURF": "WCU focal plane mask",   # TODO more specific?
+           "CTYPE1": "LINEAR",
+           "CTYPE2": "LINEAR",
+           "CRPIX1": 1024.5,
+           "CRPIX2": 1024.5,
+           "CRVAL1": 0.,
+           "CRVAL2": 0.,
+           "CUNIT1": "arcsec",
+           "CUNIT2": "arcsec",
+           "CDELT1": 0.00547,
+           "CDELT2": 0.00547,
+           "BUNIT": "PHOTLAM arcsec-2",
+           "SOLIDANG": "arcsec-2"}
+
+    def __init__(self,
+                 maskname: Path | str | None = None,
+                 fpmask_filename_format: str | None = None,
+                 angle: float = 0,
+                 shift: tuple = (0, 0),
+                 **kwargs
+                 ):
+        logger.debug("Initialising FPMask with {}".format(maskname))
```

Comment:
```suggestion
        logger.debug("Initialising FPMask with %s", maskname)
```
(that way the formatting is only executed when debug logging is actually on)